tensorflow/lite/micro/kernels/unidirectional_sequence_lstm_test.cc

```diff
@@ -918,198 +2946,2618 @@ void TestUnidirectionalSequenceLstmInteger(
+      tflite::testing::
+          recurrent_to_output_w_no_cifg_peephole_proj_clipping_scale,
+      tflite::testing::recurrent_to_output_w_no_cifg_peephole_proj_clipping_zp,
+      &tflite::testing::
+          recurrent_to_output_w_no_cifg_peephole_proj_clipping_qparam,
+
+      tflite::testing::cell_to_input_w_no_cifg_peephole_proj_clipping,
+      tflite::testing::cell_to_input_w_no_cifg_peephole_proj_clipping_quant,
+      tflite::testing::cell_to_input_w_no_cifg_peephole_proj_clipping_scale,
+      tflite::testing::cell_to_input_w_no_cifg_peephole_proj_clipping_zp,
+      &tflite::testing::cell_to_input_w_no_cifg_peephole_proj_clipping_qparam,
+
+      tflite::testing::cell_to_forget_w_no_cifg_peephole_proj_clipping,
+      tflite::testing::cell_to_forget_w_no_cifg_peephole_proj_clipping_quant,
+      tflite::testing::cell_to_forget_w_no_cifg_peephole_proj_clipping_scale,
+      tflite::testing::cell_to_forget_w_no_cifg_peephole_proj_clipping_zp,
+      &tflite::testing::cell_to_forget_w_no_cifg_peephole_proj_clipping_qparam,
+
+      tflite::testing::cell_to_output_w_no_cifg_peephole_proj_clipping,
+      tflite::testing::cell_to_output_w_no_cifg_peephole_proj_clipping_quant,
+      tflite::testing::cell_to_output_w_no_cifg_peephole_proj_clipping_scale,
+      tflite::testing::cell_to_output_w_no_cifg_peephole_proj_clipping_zp,
+      &tflite::testing::cell_to_output_w_no_cifg_peephole_proj_clipping_qparam,
+
+      tflite::testing::input_gate_bias_no_cifg_peephole_proj_clipping,
+      tflite::testing::forget_gate_bias_no_cifg_peephole_proj_clipping,
+      tflite::testing::cell_gate_bias_no_cifg_peephole_proj_clipping,
+      tflite::testing::output_gate_bias_no_cifg_peephole_proj_clipping,
+
+      tflite::testing::projection_w_no_cifg_peephole_proj_clipping,
+      tflite::testing::projection_w_no_cifg_peephole_proj_clipping_quant,
+      tflite::testing::projection_w_no_cifg_peephole_proj_clipping_scale,
+      tflite::testing::projection_w_no_cifg_peephole_proj_clipping_zp,
+      &tflite::testing::projection_w_no_cifg_peephole_proj_clipping_qparam,
+
+      /*projection_bias=*/nullptr,
+
+      tflite::testing::output_state_no_cifg_peephole_proj_clipping,
+      tflite::testing::cell_state_no_cifg_peephole_proj_clipping,
+
+      /*input_layer_norm_coefficients=*/nullptr,
+      /*forget_layer_norm_coefficients=*/nullptr,
+      /*cell_layer_norm_coefficients=*/nullptr,
+      /*output_layer_norm_coefficients=*/nullptr,
+
+      tflite::testing::output_no_cifg_peephole_proj_clipping,
+      tflite::testing::golden_output_no_cifg_peephole_proj_clipping_original,
+      tflite::testing::golden_output_no_cifg_peephole_proj_clipping,
+
+      /*tolerance=*/0.00467,
+      /*input_output_batch_major=*/false,
+      /*weight_type=*/kTfLiteUInt8);
+}
+
+TF_LITE_MICRO_TEST(UndrctnlSqncLstmFloatNoCifgPphlPrjBiasClpTest) {
+  tflite::testing::TestUnidirectionalSequenceLstmFloat(
```

Comment:
just thinking outloud, what is your opinion of defining a TestConfig struct and pass in that test config to the test function? Would that make it better to read? I feel too many input arguments are difficult to read, but also recognize there may be trade-off here

Reply:
Thanks. Great suggestion. I will change this PR to draft and refactor.